### PR TITLE
bug 1711207 clarifying compute/control plane types

### PR DESCRIPTION
**Pull request by kalexand-rh**
Draft of https://bugzilla.redhat.com/show_bug.cgi?id=1711207

@morenod, @wking, will you PTAL?

### CODE REVIEW

File: modules/installation-aws-config-yaml.adoc [hunk 1]
```diff
@@ -73,12 +73,19 @@ platform:
...
 pullSecret: '{"auths": ...}' <1>
-sshKey: ssh-ed25519 AAAA... <3>
+sshKey: ssh-ed25519 AAAA... <5>
 ----
 <1> Required. The installation program prompts you for this value.
 <2> If you do not provide these parameters and values, the installation program
 provides the default value.
-<3> You can optionally provide the `sshKey` value that you use to access the
-machines in your cluster.
+<3> The `controlplane` section is a singlet, and the `compute` section is an
```
Comment (wking) [inline modules/installation-aws-config-yaml.adoc]: `controlplane` -> `controlPlane` throughout

File: modules/installation-aws-config-yaml.adoc [hunk 2]
```diff
@@ -73,12 +73,19 @@ platform:
...
+array. Because the sections have different definitions, the first line of the
+`compute` section must begin with a hyphen, `-`, and the first line of the
+`controlplane` section must not. Although both sections currently define a
+single machine pool, it is possible that future versions of {product-title}
+will support defining multiple compute pools during installation. Only one
+control plane pool is used in the current system design.
```
Comment (wking) [inline modules/installation-aws-config-yaml.adoc]: I dunno if we need the "current design" wiggle, but ask the master team to confirm.

File: modules/installation-aws-config-yaml.adoc [hunk 3]
```diff
@@ -73,12 +73,19 @@ platform:
...
 <4> To configure faster storage for etcd, especially for larger clusters, set the
 storage type as `io1` and set `iops` to `2000`.
+<5> You can optionally provide the `sshKey` value that you use to access the
```
Comment (wking) [inline modules/installation-aws-config-yaml.adoc]: Strengthen to "you need to set this if you want to debug several classes of install failures" or some such.

File: modules/installation-aws-config-yaml.adoc (hunk 2 above)
Comment (kalexand-rh) [reply modules/installation-aws-config-yaml.adoc]: I'll just take out the wiggle.

File: modules/installation-aws-config-yaml.adoc (hunk 3 above)
Comment (kalexand-rh) [reply modules/installation-aws-config-yaml.adoc]: I'll add it to https://github.com/openshift/openshift-docs/pull/14766/.

File: modules/installation-bare-metal-config-yaml.adoc [hunk 4]
```diff
@@ -14,39 +14,46 @@ parameters.
...
+<5> The cluster name.
+<6> A block of IP addresses from which pod IPs are allocated.
+<7> The subnet prefix length to assign to each individual node. For example, if
 `hostPrefix` is set to `23`, then each node is assigned a `/23` subnet out of
 the given `cidr`, which allows for 510 (2^(32 - 23) - 2) pod IPs addresses.
-<7> The IP address pool to use for service IP addresses. You can enter only
+<8> The IP address pool to use for service IP addresses. You can enter only
 one IP address pool.
-<8> The pull secret that provides components in the cluster access to images for
+<9> The pull secret that provides components in the cluster access to images for
```
Comment (DanyC97) [inline modules/installation-bare-metal-config-yaml.adoc]: @kalexand-rh  @wking my 0.02 $: for folks who are new on v4 or those coming from v3, i think will be useful to reword it to explicitly say what this pullSecret is, something like 

```
The pull secret that provides components in the cluster access to container images for {product-title} components
```
i think will do.

I'd even go and maybe extend the pullSecret glossary to say that is nothing but the login credentials to a docker private registry which is holding the container images

Comment (wking) [reply modules/installation-bare-metal-config-yaml.adoc]: "images" -> "container images" is fine with me.  And that's the main pull-secret goal, but it also authorizes you to telemetry, and maybe other things I'm not aware of.  So in anything as global as a glossary, I'd probably say that it's about authenticating your cluster as belonging to you for services provided by the included authorities, including quay.io for serving container images.

Comment (kalexand-rh) [reply modules/installation-bare-metal-config-yaml.adoc]: @wking, I updated the pull secret description. What do you think?

Comment (wking) [reply modules/installation-bare-metal-config-yaml.adoc]: Looks great

Comment (DanyC97) [reply modules/installation-bare-metal-config-yaml.adoc]: looks great, thanks @kalexand-rh @wking !

**Comment by kalexand-rh** [comment]
This change was approved on the source bug. Merging.
